starter-code/public/javascript/index.js

```diff
@@ -1,23 +1,141 @@
+        </div>`
+        });
+        const container = document.querySelector(".character-info");
+        container.innerHTML = tpl;
+      })
+      .catch(apiErr => {
+        console.log(apiErr);
+      });
+
   });
 
   document.getElementById('fetch-one').addEventListener('click', function (event) {
 
+    const inputId = document.getElementById("fetch-one-input");
+    const id = inputId.value;
+
+    console.log(id);
+
+    charactersAPI
+      .getOneRegister(id)
+      .then(apiRes => {
+        const character = apiRes.data;
+        let tpl = "";
+        tpl += `<div class="character-info">
+        <div class="name">Id: <span>${character.id}</span></div>
+        <div class="name">Name: <span>${character.name}</span></div>
+        <div class="occupation">Occupation: <span>${character.occupation}</span></div>
+        <div class="cartoon">Cartoon: <span>${character.cartoon}</span></div>
+        <div class="weapon">Weapon: <span>${character.weapon}</span></div>
+          </div>`
+
+        const container = document.querySelector(".character-info");
+        container.innerHTML = tpl;
+      })
+      .catch(apiErr => {
+        console.log(apiErr);
+      });
+
   });
 
   document.getElementById('delete-one').addEventListener('click', function (event) {
 
+    const inputId = document.getElementById("delete-one-input");
+    const id = inputId.value;
+    const deleteButton = document.getElementById("delete-one");
+
+    charactersAPI
+      .deleteOneRegister(id)
+      .then(apiRes => {
+        deleteButton.classList.add("active");
+        deleteButton.classList.remove("wrong");
+      })
+      .catch(apiErr => {
+        deleteButton.classList.add("wrong");
+        deleteButton.classList.remove("active");
+      });
+
   });
 
   document.getElementById('edit-character-form').addEventListener('submit', function (event) {
 
+    const updateButton = document.getElementById("create-button");
+
+    event.preventDefault();
+
+    const id = document.getElementById("edit-id").value;
+    const name = document.getElementById("edit-name").value;
+    const occupation = document.getElementById("edit-occupation").value;
+    const weapon = document.getElementById("edit-weapon").value;
+    const isACartoon = document.getElementById("edit-cartoon").checked;
+
+
+    const infos = {
+      id: id,
+      name: name,
+      occupation: occupation,
+      weapon: weapon,
+      cartoon: isACartoon
+    }
+
+    charactersAPI.updateOneRegister(id, infos)
+      .then(apiRes => {
+        updateButton.classList.add("active");
+        updateButton.classList.remove("wrong");
+      })
+      .catch(apiErr => {
+        updateButton.classList.add("wrong");
+        updateButton.classList.remove("active");
+      })
+
   });
 
   document.getElementById('new-character-form').addEventListener('submit', function (event) {
 
+    const createButton = document.getElementById("create-button");
```

Comment:
be carefull, there is a mistake in the html because 2 buttons have same id ("create-button")
So when you have an error on the update, it's the new character button that change color as it is first in the html node.
So just change the id in the html ;)

Reply:
Also, check if there is empty input in the create and update ? Otherwise you will have empty properties inside your character